### PR TITLE
Pass operator to rule value renderer

### PR DESCRIPTION
**Pull request by simasjar**
closes #iTwin/presentation/issues/175
## Changes

- Promoted `PropertyFilterBuilderRuleOperatorProps`, `PropertyFilterBuilderRuleValueProps`, `PropertyFilterBuilderRuleValue` to beta, created interface `PropertyFilterBuilderRuleValueRendererProps` which extends `PropertyFilterBuilderRuleValueProps`, passed `PropertyFilterRuleOperator` to custom `PropertyFilterBuilderRuleValue` renderer.
- Minor UI fixes

## Testing
N/A

### CODE REVIEW

**Review by saskliutas — CHANGES_REQUESTED** [review]

File: ui/components-react/src/components-react/filter-builder/FilterBuilderRuleValue.tsx [hunk 1]
```diff
@@ -11,10 +11,10 @@ import type { PropertyDescription, PropertyValue } from "@itwin/appui-abstract";
 import { PropertyRecord, PropertyValueFormat } from "@itwin/appui-abstract";
 import type { PropertyUpdatedArgs } from "../editors/EditorContainer";
 import { EditorContainer } from "../editors/EditorContainer";
+import { PropertyFilterRuleOperator } from "./Operators";
 
 /**
  * Props for [[PropertyFilterBuilderRuleValue]] component.
- * @internal
```
[the remark this answers is not in the document]
Comment (simasjar) [reply ui/components-react/src/components-react/filter-builder/FilterBuilderRuleValue.tsx:17]: fixed?

File: ui/components-react/src/components-react/filter-builder/FilterBuilderRule.scss [hunk 2]
```diff
@@ -10,8 +10,14 @@
 
   .rule-condition {
     display: grid;
-    grid-template-columns: minmax(auto, 500px) 150px minmax(auto, 300px);
+    grid-template-columns: minmax(auto, 500px) 150px minmax(auto, 100%);
     grid-column-gap: var(--iui-size-m);
+    width: calc(100% - var(--iui-size-s));
+
+    .rule-property > .iui-input-container > .iui-input-with-icon > .iui-input,
```
Comment (saskliutas) [inline ui/components-react/src/components-react/filter-builder/FilterBuilderRule.scss:17]: Why these styles were added?

File: common/api/summary/components-react.exports.csv
Comment (raplemie) [inline common/api/summary/components-react.exports.csv:1]: There shouldnt be as many changes in that file, try resetting it, building and re running `npm run extract-api` in the `components-react` folder, there should be less changes.

**Comment by raplemie** [comment]
Please add relevant comment in NextVersion.md

File: common/changes/@itwin/components-react/pass_operator_to_rule_value_renderer_2023-07-19-13-26.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/components-react",
+      "comment": "Promote `PropertyFilterBuilderRuleValueProps` and `PropertyFilterBuilderRuleOperatorProps`  to beta, `PropertyFilterBuilderRuleValue` to public and pass `PropertyFilterRuleOperator` to `PropertyFilterBuilderRuleValue`.",
```
[the remark this answers is not in the document]
Comment (simasjar) [reply common/changes/@itwin/components-react/pass_operator_to_rule_value_renderer_2023-07-19-13-26.json]: updated

File: common/api/summary/components-react.exports.csv
Comment (simasjar) [reply common/api/summary/components-react.exports.csv:1]: fixed

**Comment by simasjar** [comment]
> Please add relevant comment in NextVersion.md

added

**Review by saskliutas — APPROVED** [review]

File: ui/components-react/src/components-react/filter-builder/FilterBuilderRuleOperator.tsx [hunk 4]
```diff
@@ -17,7 +17,7 @@ import {
 
 /**
  * Props for [[PropertyFilterBuilderRuleOperator]] component.
- * @internal
+ * @beta
```
Comment (GerardasB) [inline ui/components-react/src/components-react/filter-builder/FilterBuilderRuleOperator.tsx:20]: This module is not re-exported in a barrel, the API extractor doesn't pick it up.

Comment (simasjar) [reply ui/components-react/src/components-react/filter-builder/FilterBuilderRuleOperator.tsx:20]: fixed

File: ui/components-react/src/components-react/filter-builder/FilterBuilder.tsx [hunk 5]
```diff
@@ -26,6 +26,17 @@ import {
   usePropertyFilterBuilderState,
 } from "./FilterBuilderState";
 import type { PropertyFilter } from "./Types";
+import type { PropertyFilterRuleOperator } from "./Operators";
+
+/**
+ * Props for custom ruleValueRenderer.
+ * @beta
+ */
+export interface CustomPropertyFilterBuilderRuleValueProps
```
Comment (saskliutas) [inline ui/components-react/src/components-react/filter-builder/FilterBuilder.tsx:35]: I think this type should be defined in `FilterBuilderRuleValue` file and maybe rename it to `PropertyFilterBuilderRuleValueRendererProps`

Comment (simasjar) [reply ui/components-react/src/components-react/filter-builder/FilterBuilder.tsx]: fixed

File: ui/components-react/src/components-react/filter-builder/FilterBuilderRuleValue.tsx [hunk 6]
```diff
@@ -25,6 +26,16 @@ export interface PropertyFilterBuilderRuleValueProps {
   onChange: (value: PropertyValue) => void;
 }
 
+/**
+ * Props for custom ruleValueRenderer.
```
Comment (saskliutas) [inline ui/components-react/src/components-react/filter-builder/FilterBuilderRuleValue.tsx:30]:
```suggestion
 * Props for custom [[PropertyFilterBuilderRuleValue]] renderer.
```

**Review by GerardasB — APPROVED** [review]

**Pull request merged**